packages/plugin-parser/src/ParsingEngine.ts

```diff
@@ -11,18 +11,21 @@ export interface Parser<T = {}> {
 
 export class ParsingEngine<T = {}> {
     static readonly id: ParsingIdentifier;
-    static defaultParser: Constructor<Parser>;
+    static readonly defaultParser: ParserConstructor;
+    static fallbacks?: ParsingIdentifier[];
```

Comment:
I would put `fallbacks` (or `extends`) right after `id`. In addition, I would put it `readonly`.

Reply:
I would also put back the `Constructor<Parser>` type.